pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -77,58 +75,21 @@ const MasterVMNamePrefix = "k8s-master-"
 const MasterPoolName = "master"
 
 // UpgradeCluster runs the workflow to upgrade a Kubernetes cluster.
-func (uc *UpgradeCluster) UpgradeCluster(subscriptionID uuid.UUID, az armhelpers.AKSEngineClient, kubeConfig, resourceGroup string,
-	cs *api.ContainerService, nameSuffix string, agentPoolsToUpgrade []string, aksEngineVersion string) error {
-	uc.ClusterTopology = ClusterTopology{}
-	uc.SubscriptionID = subscriptionID.String()
-	uc.ResourceGroup = resourceGroup
-	uc.DataModel = cs
-	uc.NameSuffix = nameSuffix
+func (uc *UpgradeCluster) UpgradeCluster(az armhelpers.AKSEngineClient, kubeConfig string, aksEngineVersion string) error {
 	uc.MasterVMs = &[]compute.VirtualMachine{}
 	uc.UpgradedMasterVMs = &[]compute.VirtualMachine{}
 	uc.AgentPools = make(map[string]*AgentPoolTopology)
-	uc.AgentPoolsToUpgrade = make(map[string]bool)
 
-	for _, poolName := range agentPoolsToUpgrade {
-		uc.AgentPoolsToUpgrade[poolName] = true
-	}
-	uc.AgentPoolsToUpgrade[MasterPoolName] = true
-
-	if err := uc.getClusterNodeStatus(subscriptionID, az, resourceGroup, kubeConfig); err != nil {
+	if err := uc.getClusterNodeStatus(az, uc.ResourceGroup, kubeConfig); err != nil {
 		return uc.Translator.Errorf("Error while querying ARM for resources: %+v", err)
 	}
 
 	var upgrader UpgradeWorkFlow
 	upgradeVersion := uc.DataModel.Properties.OrchestratorProfile.OrchestratorVersion
 	uc.Logger.Infof("Upgrading to Kubernetes version %s\n", upgradeVersion)
-	switch {
-	case strings.HasPrefix(upgradeVersion, "1.6."):
```

Comment:
if anyone knows a good reason why these separate upgraders are still needed, please speak up

Reply:
:+1: as far as I can tell, these structs provide no utility over the `Upgrader` base.

Maybe someone was anticipating we would need different behavior for each version upgrade? (Don't do that in advance because [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).)